### PR TITLE
Fix some typos and improve copy

### PR DESCRIPTION
**Pull request by louim**
Hello!

I fixed some typos in the readme, and improved some comments. Cheers! :smile:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,24 +46,24 @@ challenge.filename # => ".well-known/acme-challenge/:some_token"
 # You can generate the body of the expected response.
 challenge.file_content # => 'string token and JWK thumbprint'
 
-# You can send no Content-Type at all but if you send one it has to be 'text/plain'.
+# You are not required to send a  Content-Type at all. If you do send one it has to be 'text/plain'.
```
Comment (jhass) [inline README.md]: "... it has to be the one returned from this method." ? Else this comment kinda makes the entire method redundant.

File: README.md [hunk 2]
```diff
@@ -32,10 +35,10 @@ registration.agree_terms
 # We need to prove that we control the domain using one of the challenges method.
 authorization = client.authorize(domain: 'example.org')
 
-# For now the only challenge method supprted by the client is http-01.
+# For now the only challenge method supported by this client is http-01.
```
Comment (jhass) [inline README.md]: This is actually outdated.

Comment (louim) [inline README.md]: I did not dig into the code, can you tell me what other method are supported? Would it make sense to update the readme with example from other methods?

File: README.md (hunk 1 above)
Comment (louim) [inline README.md]: You are right, that didn't make much sense! Fixed.

File: README.md (hunk 2 above)
Comment (jhass) [inline README.md]: https://github.com/unixcharles/acme-client/blob/master/lib/acme/client/resources/authorization.rb#L19-L21

I'm not sure I'd add more examples here, it's probably time for some more structured documentation than a single big example.

Comment (unixcharles) [inline README.md]: We have full support for all methods implemented in Boulder, so I think we could just remove the mention.

The only missing challenge from the acme spec is proof-of-possession but I prefer not add it until a server side implementation is completed.

> I'm not sure I'd add more examples here, it's probably time for some more structured documentation than a single big example.
 
I agree, we could start linking to the wiki or create an example/ directory.

**Comment by louim** [comment]
Removed the comment about the only type of challenge supported, rebased against current master and squashed. :+1:

**Comment by unixcharles** [comment]
awesome! thanks @louim

**Pull request merged**